src/sgame/sg_cmds.cpp

```diff
@@ -3497,6 +3497,97 @@ void Cmd_Reload_f( gentity_t *ent )
 	}
 }
 
+void Cmd_TeamStatus_f( gentity_t * ent )
+{
+	int builders = 0;
+	gentity_t *tmp;
+	struct buildable_status_t
+	{
+		int count;
+		int health;
+		bool spawned;
+	} structures[ BA_NUM_BUILDABLES ] = {};
+
+	if ( g_teamStatus.Get() <= 0 ) 
+	{
+		trap_SendServerCommand( ent - g_entities, "print \"teamstatus is disabled.\n\"" );
+		return;
+	}
+
+	if ( ent->client->pers.namelog->muted ) 
+	{
+		trap_SendServerCommand( ent - g_entities, "print \"You are muted and cannot use message commands.\n\"" );
+		return;
+	}
+
+	if ( level.team[ G_Team( ent ) ].lastTeamStatus
+		&& ( level.time - level.team[ G_Team( ent ) ].lastTeamStatus ) < g_teamStatus.Get() * 1000 ) 
+	{
+		trap_SendServerCommand( ent - g_entities, va( "print \"Your team's status may only be checked every %i seconds.\"", g_teamStatus.Get() ) );
+		return;
+	}
+
+	level.team[ G_Team( ent ) ].lastTeamStatus = level.time;
+
+	tmp = &g_entities[ 0 ];
+	for ( int i = 0; i < level.num_entities; i++, tmp++ ) 
+	{
+		if ( !G_OnSameTeam( tmp, ent ) )
+		{
+			continue;
+		}
+		
+		if ( i < MAX_CLIENTS && tmp->client && tmp->entity ) 
+		{
+			const auto& health = tmp->entity->Get<HealthComponent>();
+			if ( tmp->client->pers.connected == CON_CONNECTED
+				&& ( health && health->Alive() )
+			    && ( tmp->client->ps.stats[ STAT_CLASS ] == PCL_ALIEN_BUILDER0
+					|| tmp->client->ps.stats[ STAT_CLASS ]  == PCL_ALIEN_BUILDER0_UPG
+					|| BG_InventoryContainsWeapon( WP_HBUILD, tmp->client->ps.stats ) ) )
+			{
+				builders++;
+			}
+			continue;
+		}
+
+		if ( tmp->s.eType == entityType_t::ET_BUILDABLE ) 
+		{
+			int type = tmp->s.modelindex;
+			ASSERT( type < BA_NUM_BUILDABLES );
+			const HealthComponent* health = tmp->entity->Get<HealthComponent>();
+			if ( health->Health() > 0 )
```

Comment:
I would favor this as I'm not 100% sure every buildabe has a health component all the time
```suggestion
			if ( health && health->Health() > 0 )
```